src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -954,6 +934,33 @@ internal static void PerformWaitOrTimerCallback(_ThreadPoolWaitOrTimerCallback h
 
     public static partial class ThreadPool
     {
+        internal static readonly ThreadPoolWorkQueue s_workQueue = new ThreadPoolWorkQueue();
+        internal static readonly bool s_enableWorkerTracking = GetEnableWorkerTracking();
```

Comment:
@kouvel, any reason this is a problem?  The code was going out of its way before to not do this, and I'm wondering why.

Reply:
I'm not sure why it was not done this way before. It looks like most of the public APIs on `ThreadPool` would initialize the thread pool either from managed code or from the unmanaged side. At the moment tiering would initialize it early anyway. I don't see a problem with this.

Reply:
I assume it will be better for Mono AoT also since they are just constants there and can be eliminated entirely (as well as eliminating them in Core at Tier1)